module_3/gift_certificates_system/repository/src/main/java/com/epam/esm/dao/GiftCertificateToTagRelationDao.java

```diff
@@ -0,0 +1,89 @@
+package com.epam.esm.dao;
+
+import com.epam.esm.entity.GiftCertificate;
+import com.epam.esm.entity.GiftCertificateToTagRelation;
+import com.epam.esm.entity.Tag;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Gift certificate to tag relation DAO layer class.<br>
+ * Works with database.
+ */
+public interface GiftCertificateToTagRelationDao<T> {
+    /**
+     * Create gift certificate to tag relation by gift certificate and tag.
+     *
+     * @param gc  Gift certificate.
+     * @param tag Tag.
+     * @return Created gift certificate to tag relation entity.
+     */
+    GiftCertificateToTagRelation create(GiftCertificate gc, Tag tag);
+
+    /**
+     * Create gift certificate to tag relation.
+     *
+     * @param relation Gift certificate to tag relation entity.
+     * @return Created gift certificate to tag relation entity.
+     */
+    GiftCertificateToTagRelation create(GiftCertificateToTagRelation relation);
```

Comment:
Similar

Reply:
Abbreviations were used to avoid long names.

Reply:
@n199a it's not about naming, but about almost similar methods for relation creation. Probably, you can get rid of one of these

Reply:
> @n199a it's not about naming, but about almost similar methods for relation creation. Probably, you can get rid of one of these
Fixed.